taskweaver/code_interpreter/code_interpreter_plugin_only.py

```diff
@@ -0,0 +1,91 @@
+import json
+from typing import Optional
+
+from injector import inject
+
+from taskweaver.code_interpreter.code_executor import CodeExecutor
+from taskweaver.code_interpreter.code_generator import CodeGeneratorPluginOnly
+from taskweaver.config.module_config import ModuleConfig
+from taskweaver.logging import TelemetryLogger
+from taskweaver.memory import Memory, Post
+from taskweaver.memory.attachment import AttachmentType
+from taskweaver.role import Role
+
+
+class CodeInterpreterConfig(ModuleConfig):
+    def _configure(self):
+        self._set_name("code_interpreter_plugin_only")
+        self.use_local_uri = self._get_bool("use_local_uri", False)
+        self.max_retry_count = self._get_int("max_retry_count", 3)
+
+
+class CodeInterpreterPluginOnly(Role):
```

Comment:
Just wondering why we need to have a separate code interpreter rather than just using the existing one considering the code generation logic should mostly in the code generator?

Reply:
It is quite different. Adding the new logic to the original class just makes it messy and hard to maitain in the future. This new one is much simplier.